fdbrpc/BlobStore.actor.cpp

```diff
@@ -874,6 +898,8 @@ Future<int> BlobStoreEndpoint::readObject(std::string const &bucket, std::string
 }
 
 ACTOR static Future<std::string> beginMultiPartUpload_impl(Reference<BlobStoreEndpoint> bstore, std::string bucket, std::string object) {
+	Void _ = wait(bstore->requestRateWrite->getAllowance(1));
```

Comment:
So, because this inserts a request rate checker before the beginning of the multipart and before each part, that means that an upload will take (at minimum) roughly writes per second * parts seconds, right? (So, at 100 writes per second, a multipart upload with 10 parts will take at least 100 ms to complete.) In the case of many concurrent writers, this will mean that the rate limiting will be applied per part request (rather than, say, asking for `k + 2` allowances at the beginning and then just going at it). I think this is probably the right call, but I guess I wanted to make sure that that was the thinking.

Reply:
That's right, and it's because the multi part uploads are used for streaming uploads, so when the first part upload begins it isn't yet known how many parts there will be.